Added `maxWidth` prop to `EuiFlyout`

cchaos opened the pull request:
Fixes #1071

Similarly to how `EuiPage` uses the `restrictWidth` prop, the `maxWidth` prop can be:
- `false` (default): No max-width
- `true`: Adds a pre-defined max width depending on flyout `size`
- `number`: Will add the number to the style prop as a `maxWidth` value in `px`
- `string`: Will add the full string as the `maxWidth` value in whatever measurement it provides

---

The `string` option has now also been added to `EuiPage` and `EuiPageBody` as well as the simplified style js commented below, including removing the unnecessary `--widthIsNotRestricted` classes, and adding this prop to the TS definitions.

chandlerprall commented on src/components/flyout/flyout.js:
```diff
@@ -34,10 +34,28 @@ export class EuiFlyout extends Component {
+    } else if (maxWidth !== false) {
+      // if style has been passed as a prop, add to it
+      if (style) {
+        newStyle = style;
```
Best to create a new style object here,

`newStyle = { ...style, maxWidth };`

otherwise this will add the `maxWidth` key/value to the original style prop.

cchaos replied:
I also get confused about JS's stupid array reference stuff and forget about the spread method. Just so I understand, does the way your method work is that it will add the key/value pair of `maxWidth` or is it just adding that value?

cchaos replied:
Oh wow, that actually completely negates the need for the if statement, because if style is empty, then there's just no other key/value pairs. Nice!

cchaos commented:
@chandlerprall Can you take another look. I also updated `EuiPage` and `EuiPageBody` to utilize the same type of logic.

Can you also check that I added the TS stuff correctly?

chandlerprall commented:
@cchaos all looks good to me (including TS defs), thanks for the changes!
